### ISSUE TITLE
Azure Notebook Unable to Reconnect after Multiple Month Connectivity

### ISSUE THREAD

**Issue by Khalipha2019** [Microsoft/AzureNotebooks#611]
Please describe your issue as completely as possible.

### Required:  Error message: Connection failed
A connection to the notebook server could not be established. The notebook will continue trying to reconnect. Check your network connection or notebook server configuration.
#### Which site did you access us from?
    [*] https://notebooks.azure.com
    [] https://studio.azureml.net

#### Describe the issue you are encountering
Upon clicking on Library from My Projects in Azure Notebook, notebook opens 1 sec, kernel connects, and then I immediately see the error message.  I am able to open a notebook from my home environment, but it does not load csv's. Thus, I am unable to complete the online learning I am walking thru for the Microsoft Professional Program at which I am at 70% completion. 

I have completed the following troubleshooting steps:
1. Cloned the library, still unable to connect
2. Confirmed Chrome browser set as default.
3. Confirmed proxy settings disabled. 
4. Cleared space on laptop.
5. Restarted kernel several times. 
6. Updated JAVA to 130.199. from 139
7. Confirmed network discovery is on. 
8. Note: the library was cloned from GitHub, it worked for two days and then broke.
9. Did several online searches and read related GitHub and MSDN posts, to no avail. 
10. Also posted to the same question to MSDN with less detail than is presented here. 
 Any assistance you can provide would be greatly appreciated.  I am open to RDP. 
#### Which language/kernel were you using when you experienced this issue?
    [] Python 2.7 
    [] Python 3.5
    [*] Python 3.6 
    [] R 
    [] F#
    [] Terminal

#### What time (please include time zone) did this occur?
1/13/09 2:35 PM
#### Which Web Browser(s) did you see the error in?
    [*] Chrome
    [] Firefox
    [] Internet Explorer / Edge (Please note this is not supported)
    [] Other (Please Specify)

#### Which Operating System are you using? Please specify version as well if possible:
    [*] Windows
    [] MacOS
    [] Ubuntu
    [] Other (Please Specify)

#### What is the path to the notebook file (if relevant) that reproduces the issue? For example, https://notebooks.azure.com/<USER-NAME>/libraries/<LIBRARY-NAME>/html/<PATH-TO-FILE>. If this is sensitive, you can email it to nbhelp@microsoft.com
(https://principlesofmachinelearning-kenyattaazurelab.notebooks.azure.com/j/tree/Module3 [https://principlesofmachinelearning-kenyattaazurelab.notebooks.azure.com/j/notebooks/Module3/DataPreparation.ipynb]  33.6 kB 20 hours ago])
    
#### Each notebook instance has log files. It may be helpful for us to have them. You can retrieve them by following this [help article](https://notebooks.azure.com/help/jupyter-notebooks/troubleshooting/jupyter-debug-log)

Unfortunately, unable to access logs, at this time; due to:

Connection failed
A connection to the notebook server could not be established. The notebook will continue trying to reconnect. Check your network connection or notebook server configuration.

Any direct assistance aka RDP would be greatly appreciated.

**Comment by rachelgshaffer** [comment]
I'm sorry you're experiencing this issue. Is this still occurring today? Are there any errors in the dev tools console (click F12 in Chrome or Edge to open the dev tools panel). Another possible mitigation step: clearing cookies. Thanks for using Azure Notebooks!

**Comment by Khalipha2019** [comment]
Rachel,
Thanks for the reply.  There are three  errors in the Dev console..  I have enclosed links and pics.
Now the Notebook opens the page and displays the kernel as connected, but it will not run code.

Also, I have cleared cookies multiple times. 



https://notebooks.azure.com/Scripts/Jupyter/5.5.0/notebook/js/main.min.js

https://notebooks.azure.com/Scripts/Jupyter/5.5.0/notebook/js/main.min.js

https://notebooks.azure.com/Scripts/Jupyter/5.5.0/notebook/js/main.min.js



1. {readyState: 4, getResponseHeader: ƒ, getAllResponseHeaders: ƒ, setRequestHeader: ƒ, overrideMimeType: ƒ, …}
1. abort: ƒ (a)
2. always: ƒ ()
3. complete: ƒ ()
4. done: ƒ ()
5. error: ƒ ()
6. fail: ƒ ()
7. getAllResponseHeaders: ƒ ()
8. getResponseHeader: ƒ (a)
9. overrideMimeType: ƒ (a)
10. pipe: ƒ ()
11. progress: ƒ ()
12. promise: ƒ (a)
13. readyState: 4
14. responseText: "<html><head><title>500 - The request timed out.</title></head><body>  <font color ="#aa0000">         <h2>500 - The request timed out.</h2></font>  The web server failed to respond within the specified time.</body></html>"
15. setRequestHeader: ƒ (a,b)
16. state: ƒ ()
17. status: 500
18. statusCode: ƒ (a)
19. statusText: "error"
20. success: ƒ ()
1. arguments: null
2. caller: null
3. length: 0
4. name: "add"
5. prototype: {constructor: ƒ}
6. __proto__: ƒ ()
7. [[FunctionLocation]]: <unknown>
8. [[Scopes]]: Scopes[3]
21. then: ƒ ()
22. __proto__: Object





Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/



From: Laurie HK
Sent: Friday, January 18, 2019 9:34 PM
To: Microsoft/AzureNotebooks; Microsoft/AzureNotebooks
Cc: Author
Subject: RE: Additional Errors: Khalipha2019 Reply -[Microsoft/AzureNotebooks] Azure Notebook Unable to Reconnectafter MultipleMonth Connectivity (#611)




Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/



From: Laurie HK
Sent: Friday, January 18, 2019 8:35 PM
To: Microsoft/AzureNotebooks
Subject: RE: [Microsoft/AzureNotebooks] Azure Notebook Unable toReconnectafter Multiple Month Connectivity (#611)

Rachel,
Thanks for the reply.  There are three  errors in the Dev console..  I have enclosed links and pics.
Now the Notebook opens the page and displays the kernel as connected, but it will not run code.

Also, I have cleared cookies multiple times. 



https://notebooks.azure.com/Scripts/Jupyter/5.5.0/notebook/js/main.min.js

https://notebooks.azure.com/Scripts/Jupyter/5.5.0/notebook/js/main.min.js

https://notebooks.azure.com/Scripts/Jupyter/5.5.0/notebook/js/main.min.js



1. {readyState: 4, getResponseHeader: ƒ, getAllResponseHeaders: ƒ, setRequestHeader: ƒ, overrideMimeType: ƒ, …}
1. abort: ƒ (a)
2. always: ƒ ()
3. complete: ƒ ()
4. done: ƒ ()
5. error: ƒ ()
6. fail: ƒ ()
7. getAllResponseHeaders: ƒ ()
8. getResponseHeader: ƒ (a)
9. overrideMimeType: ƒ (a)
10. pipe: ƒ ()
11. progress: ƒ ()
12. promise: ƒ (a)
13. readyState: 4
14. responseText: "<html><head><title>500 - The request timed out.</title></head><body>  <font color ="#aa0000">         <h2>500 - The request timed out.</h2></font>  The web server failed to respond within the specified time.</body></html>"
15. setRequestHeader: ƒ (a,b)
16. state: ƒ ()
17. status: 500
18. statusCode: ƒ (a)
19. statusText: "error"
20. success: ƒ ()
1. arguments: null
2. caller: null
3. length: 0
4. name: "add"
5. prototype: {constructor: ƒ}
6. __proto__: ƒ ()
7. [[FunctionLocation]]: <unknown>
8. [[Scopes]]: Scopes[3]
21. then: ƒ ()
22. __proto__: Object





Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Ok. Working on that now. One sec…

Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Rachel,
Upon completion of the new import/upload from GitHub, I received the following:

 





Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Rachel,
I forgot to give the new link to the new upload of the notebook.

https://principlesofmachinelearning-kenyattaazurelab.notebooks.azure.com/j/notebooks/Module4/IntroductionToRegression.ipynb


Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by rachelgshaffer** [comment]
Thank you. In your profile settings do you have 'Unified Front End' set to on? I was able to clone the notebook and successfully run several cells. I tried shutting down and restarting a few times and it does seem to be slower to load than some other projects I have worked on, but it definitely did work. You will be able to tell it's safe to start running cells when the kernel gets to ready by hovering over this circle and it says "Kernel Idle"
![image](https://user-images.githubusercontent.com/10291901/51574564-2aca3000-1e63-11e9-8258-a179f603ff02.png)
or by seeing the message "Kernel: kernel_ready (daeb04a0-97e1-405f-aa9e-aa347beea74a)" in the dev tools console. Sometimes it may connect faster than other times. If it truly seems like nothing has happened for a minute, you can try refreshing the browser.

**Comment by Khalipha2019** [comment]
Not sure where to find Unified Front End.  Is that like a compatibility setting somewhere? I refreshed the page and got the following:



Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by rachelgshaffer** [comment]
in the upper right corner, click "My account", then "Edit profile information" then scroll to the bottom of the page.

Your message ("I refreshed the page and got the following") doesn't show anything, did you try to attach an image?

**Comment by Khalipha2019** [comment]
Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/



From: Laurie HK
Sent: Tuesday, January 22, 2019 7:59 PM
To: Microsoft/AzureNotebooks; Microsoft/AzureNotebooks
Cc: Author
Subject: RE: [Microsoft/AzureNotebooks] Azure Notebook Unable toReconnectafter Multiple Month Connectivity (#611)




Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/



From: Laurie HK
Sent: Tuesday, January 22, 2019 7:59 PM
To: Microsoft/AzureNotebooks; Microsoft/AzureNotebooks
Cc: Author
Subject: RE: [Microsoft/AzureNotebooks] Azure Notebook Unable toReconnectafter Multiple Month Connectivity (#611)




Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Rachel,
Does this help?  



Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by rachelgshaffer** [comment]
Can you email us directly at nbhelp@microsoft.com? It's very hard to follow your messages through GitHub.

**Comment by Khalipha2019** [comment]
Ok. Will resend last 3 emails sent to GitHub for your review.

Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by Khalipha2019** [comment]
Hi Rachel,
I am now actively using Microsoft Azure Jupyter Notebook.  I believe I have figured out the reason Microsoft Azure Notebook intermittently does not work with Jupyter notebook.  I have determined two methods to remove the error for future connections.  If you would like to explore the concept. Please let me know.

Laurie Kenyatta, MBA, Masters PM, ITIL, PMP
330.209.9901
https://www.linkedin.com/in/mba-mpm-itil-pmp/

**Comment by rachelgshaffer** [comment]
We recently deployed a fix that should help in kernel stability and reliability. I am going to close this issue for now, but please reactivate if you are still having trouble. Thanks for using Azure Notebooks!

**Comment by JanderHungrige** [comment]
Renew your Security application group. 
Go to your VM, Network settings, Applciation security group, Unselect the selected group and again select it. Then save.

This helped me every time I could not reconnect after a while offline.
Hope this helps.
Cheers Jan
